src/driver/mod.rs

```diff
@@ -0,0 +1,293 @@
+use bytes::BytesMut;
+use std::time::Duration;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+
+use crate::parser::{Response, Value};
+use crate::{parser, MemcacheError, PoolConnection};
+
+const EMPTY_SPACE_BYTES: &[u8] = b" ";
+const NEW_LINE_BYTES: &[u8] = b"\r\n";
+const NO_REPLY_BYTES: &[u8] = b" noreply\r\n";
+const COMMAND_DELETE: &[u8] = b"delete ";
+const COMMAND_TOUCH: &[u8] = b"touch ";
+const COMMAND_VERSION: &[u8] = b"version\r\n";
+
+/// Storage command
+#[derive(Debug)]
+pub enum StorageCommand {
+    /// "set" means "store this data".
+    Set,
+    /// "add" means "store this data, but only if the server *doesn't* already
+    /// hold data for this key".
+    Add,
+    /// "replace" means "store this data, but only if the server *does*
+    /// already hold data for this key".
+    Replace,
+}
+
+impl From<StorageCommand> for &'static [u8] {
+    fn from(c: StorageCommand) -> &'static [u8] {
+        match c {
+            StorageCommand::Set => b"set ",
```

Comment:
I noticed that `b"set ",` has whitespace after `set` but `get` command doesn't 🤔 intentionally?

Reply:
Yes, micro optimizations

Reply:
```
    // <key>
    for key in &*keys {
        let _ = conn.write(EMPTY_SPACE_BYTES).await?; // ends key without empty space
        let _ = conn.write_all(key.as_ref()).await?;
    }
    let _ = conn.write(NEW_LINE_BYTES).await?;

```

keys are a slice of strings, keys are separated by space and ends with new line. Command could have a space immediately after command, but then looping would have to have conditions to account for empty space